### PR TITLE
Fix eCAP header includes

### PR DESCRIPTION
**Pull request by yadij**
Squid style guidelines require .h files to be wrapped with
HAVE_*_H protection and placed after all Squid internal file
includes.

Add the missing ./configure header checks to generate the
needed wrappers and refactor the include sequences to meet
current guidelines.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -886,12 +886,41 @@ AS_IF([test "x$enable_ecap" != "xno"],[
...
+    libecap/common/named_values.h \
+    libecap/common/names.h \
+    libecap/common/options.h \
+    libecap/common/registry.h \
+    libecap/common/version.h \
+    libecap/host/host.h \
+    libecap/host/xaction.h \
+  ],,,[
+/* libecap-1.0.1 headers do not build without autoconf.h magic */
+#define HAVE_CONFIG_H
```
Comment (yadij) [inline configure.ac:911]: libecap apparently lacks basic self-consistency checks for its installed headers.
`AC_CHECK_HEADERS` exposes 
```
/usr/include/libecap/common/registry.h:15:75: error: 'LIBECAP_VERSION' was not declared in this scope
   15 | extern bool RegisterVersionedService(adapter::Service *s, const char *v = LIBECAP_VERSION);
```

File: configure.ac [hunk 2]
```diff
@@ -886,12 +886,41 @@ AS_IF([test "x$enable_ecap" != "xno"],[
...
+/* libecap/common/delay.h fails to include <string> */
```
Comment (yadij) [inline configure.ac:912]: libecap apparently lacks basic self-consistency checks for its installed headers.
AC_CHECK_HEADERS exposes a number of identical `std::string` errors, one example:
```
/usr/include/libecap/common/delay.h:15:43: error: 'string' in namespace 'std' does not name a type
   15 |                 explicit Delay(const std::string &state_):
      |                                           ^~~~~~
/usr/include/libecap/common/delay.h:7:1: note: 'std::string' is defined in header '<string>'; this is probably fixable by adding '#include <string>'
```

**Comment by kinkie** [comment]
May I assume that after this PR, build issues are solved?

**Comment by yadij** [comment]
Yes the commented lines are all that is necessary. These issues do no appear during the main build because we are an autoconf project and other files include `<string>`. They only show up when `./configure` checks for the headers usability.

**Comment by rousskov** [comment]
> May I assume that after this PR, build issues are solved?

What build issues? PR description does not mention any fixed build issues, and the PR has only one commit. What am I missing?

**Comment by yadij** [comment]
> > May I assume that after this PR, build issues are solved?
> 
> What build issues? PR description does not mention any fixed build issues, and the PR has only one commit. What am I missing?

The code comments documenting why the `AC_CHECK_HEADER` hacks are present.

**Comment by rousskov** [comment]
> > > May I assume that after this PR, build issues are solved?
> > 
> > 
> > What build issues? PR description does not mention any fixed build issues, and the PR has only one commit. What am I missing?
> 
> The code comments documenting why the `AC_CHECK_HEADER` hacks are present.

I was aware of those comments when I asked the question. I am interpreting your response as "The 'build issues' Francesco is talking about are not present in master/v7 code". Please correct me if I am wrong.

**Comment by yadij** [comment]
Marking for merge under the 10+ day criteria.

**Pull request closed without merging**